### PR TITLE
Backport: MM-52476 Fix guest users access to playbooks. (#23279)

### PR DESCRIPTION
**Pull request by crspeller**
## Summary

Backport of https://github.com/mattermost/mattermost/pull/23279

### CODE REVIEW

File: server/app/permissions_service.go [hunk 1]
```diff
@@ -66,10 +66,13 @@ func (p *PermissionsService) getPlaybookRole(userID string, playbook Playbook) [
 
 	// Public playbooks
 	if playbook.Public {
-		if playbook.DefaultPlaybookMemberRole == "" {
-			return []string{playbook.DefaultPlaybookMemberRole}
+		// Public playbooks are public to those who can list channels on a team. (Not guests)
+		if p.pluginAPI.User.HasPermissionToTeam(userID, playbook.TeamID, model.PermissionListTeamChannels) {
```
Comment (lieut-data) [inline server/app/permissions_service.go:70]: Why not another `guest` check vs. inferring something from a semi-related permission?

Comment (crspeller) [reply server/app/permissions_service.go:70]: Because hard guest checks are cheating vs using the permissions system properly.

Comment (lieut-data) [reply server/app/permissions_service.go:70]: In general, I agree, but using it properly would be to have a `model.PermissionViewPublicPlaybooks` or some such. This just ties together two things that happen to be related, but who knows how the product will change in the future? At least the hard-coded guest check is unlikely to regress.

Comment (crspeller) [reply server/app/permissions_service.go:70]: That's true. We should add the `model.PermissionListPublicPlaybooks` or some such. We can fix this on master. Not going to fix here for the backport.

**Pull request merged**